core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -127,7 +132,8 @@ KOKKOS_INLINE_FUNCTION
     std::enable_if_t<std::is_same<T, unsigned long long>::value, T>
         cast_from_half(half_t);
 
-class half_t {
+// use alignas(4) to support 32-bit alignment for reductions
+class alignas(4) half_t {
```

Comment:
do you really want this? That means you WILL use full 4 bytes for storage always, and will take that much memory transfer too.

Reply:
No but this is required due to the Cuda reduction backend only supporting scalar types that are multiples of 4 bytes. See https://github.com/kokkos/kokkos/pull/4156.

Reply:
@crtrott: Do you want me to close #4101 and continue work #4156?